core-rust/node-common/src/config/limits.rs

```diff
@@ -0,0 +1,109 @@
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+use radix_engine_constants::{
+    DEFAULT_MAX_SUBSTATE_READS_PER_TRANSACTION, DEFAULT_MAX_SUBSTATE_WRITES_PER_TRANSACTION,
+};
+
+// TODO: revisit & tune before Babylon
+const DEFAULT_MAX_TOTAL_TRANSACTIONS_COUNT: usize = 10;
+const DEFAULT_MAX_TOTAL_TRANSACTIONS_SIZE: usize = 4 * 1024 * 1024;
+const DEFAULT_MAX_TOTAL_VERTEX_EXECUTION_COST_UNITS_CONSUMED: usize = 200_000_000;
+const DEFAULT_MAX_TOTAL_VERTEX_SUBSTATE_READ_SIZE: usize = 40 * 1024 * 1024;
+const DEFAULT_MAX_TOTAL_VERTEX_SUBSTATE_READ_COUNT: usize =
+    DEFAULT_MAX_SUBSTATE_READS_PER_TRANSACTION * 10;
+const DEFAULT_MAX_TOTAL_VERTEX_SUBSTATE_WRITE_SIZE: usize = 10 * 1024 * 1024;
+const DEFAULT_MAX_TOTAL_VERTEX_SUBSTATE_WRITE_COUNT: usize =
+    DEFAULT_MAX_SUBSTATE_WRITES_PER_TRANSACTION * 10;
+
+pub struct VertexLimitsConfig {
+    pub max_total_transactions_count: usize,
+    pub max_total_transactions_size: usize,
+    pub max_total_execution_cost_units_consumed: usize,
+    pub max_total_substate_read_size: usize,
+    pub max_total_substate_read_count: usize,
+    pub max_total_substate_write_size: usize,
+    pub max_total_substate_write_count: usize,
+}
+
+impl VertexLimitsConfig {
```

Comment:
I'm surprised not to see a separate ProposalLimitsConfig

Reply:
Was there a separate PR for proposal limits that I missed?

Reply:
It is already on the Java side. I will capture it on the Rust side in a future refactor (and pass to Java).